var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -11,13 +12,15 @@ class Netcdf(Package):
     version('4.4.0', 'f01cb26a0126dd9a6224e76472d25f6c')
     version('4.3.3', '5fbd0e108a54bd82cb5702a73f56d2ae')
 
+    variant('mpi', default=True, description='Enables MPI parallelism')
     variant('fortran', default=False, description="Download and install NetCDF-Fortran")
     variant('hdf4',    default=False, description="Enable HDF4 support")
 
     # Dependencies:
     depends_on("curl")  # required for DAP support
     depends_on("hdf", when='+hdf4')
-    depends_on("hdf5")  # required for NetCDF-4 support
+    depends_on("hdf5+mpi~cxx", when='+mpi')  # required for NetCDF-4 support
```

Comment:
@alalazo Is it necessary to disable C++ support for HDF5 when building NetCDF 4? To my knowledge, the C++ and the C headers don't conflict, so it shouldn't matter whether C++ support in HDF5 has been enabled.

It seems this is currently causing a rat's tail of problems, amplified by an issue in Spack, that make it impossible to express the HDF5 version constraint (< 1.10) here. It seems (but I haven't tested this yet) that, without the `~cxx` specifier, the respective work-around in the HDF5 package could be avoided.

If the `~cxx` is indeed necessary here, could you then add a comment explaining why?